sympy/core/basic.py

```diff
@@ -1034,21 +1035,23 @@ def subs(self, *args, **kwargs):
         from .containers import Dict
         from .symbol import Dummy, Symbol
         from .numbers import _illegal
+        from .relational import Equality
 
         unordered = False
         if len(args) == 1:
-
             sequence = args[0]
             if isinstance(sequence, set):
                 unordered = True
             elif isinstance(sequence, (Dict, Mapping)):
                 unordered = True
                 sequence = sequence.items()
+            elif isinstance(sequence, Equality):
+                sequence = sequence.args
             elif not iterable(sequence):
                 raise ValueError(filldedent("""
                    When a single argument is passed to subs
-                   it should be a dictionary of old: new pairs or an iterable
-                   of (old, new) tuples."""))
+                   it should be either an Equality, a dictionary of old: new pairs,
```

Comment:
```suggestion
                   it should be either an Equality, a dictionary of {old: new} items,
```

Reply:
Fixing other issues in the code should be its own PR.